src/starwars-characters.js

```diff
@@ -0,0 +1,35 @@
+import React, {useState} from "react";
+import UrlDataFetcher from "./UrlDataFetcher";
+import "./style.css";
+
+export default function StarwarsCharacters({characterId, setFilmId}) {
+    const [data, setData] = useState(null);
+
+    const getCharacterDataUrl = (characterId) => `https://swapi.dev/api/people/${characterId}/`;
+
+    function extractFilmId(filmUrl) {
+        return filmUrl.match(/\/(\d+)\//)[1];
+    }
+
+    function filmDataView() {
+        return (
+            data &&
```

Comment:
You have the condition for `data` repeating twice.
It will be better to merge `filmDataView()` with the main return statement.